### ISSUE TITLE
remove dependency on /sbin/ip binary

### ISSUE THREAD

**Issue by AkihiroSuda** [rootless-containers/rootlesskit#36]

**Comment by Zyqsempai** [comment]
@AkihiroSuda Is this still actual?

**Comment by AkihiroSuda** [comment]
yes, we should still do this ideally (but tricky because of nsenter stuff)

**Comment by Zyqsempai** [comment]
@AkihiroSuda What do you think about using those packages:
For `netlink` operations we can use https://godoc.org/github.com/vishvananda/netlink
and for `nsenter` https://godoc.org/github.com/kata-containers/runtime/virtcontainers/pkg/nsenter

By the end we will get something like this
```
funcToRun := func() error {
	err := netlink.LinkSetUp(tap)
	if err != nil {
		return err
	}

	return nil
}

err = nsenter.NsEnter(nsList, funcToRun)
```

**Comment by AkihiroSuda** [comment]
You can't nsenter userns in Go.

**Comment by AkihiroSuda** [comment]
So, we should have IPC across the parent and the child process to remove nsenter dependency.
